site/en/fugu-showcase/fugu-showcase.njk

```diff
@@ -0,0 +1,245 @@
+      const matchingSearchArticles = container.querySelectorAll(`article[data-name*="${search}"]`);
+
+      matchingSearchArticles.forEach((article) => {
+        if (!matchingFilterArticles.length) article.style.display = 'block';
+        else {
+          matchingFilterArticles.forEach((filterArticle) => {
+            if (article === filterArticle) {
+              filterArticle.style.display = 'block';
+            }
+          });
+        }
+      });
+      
+      matchingFilterArticles.forEach((article) => {
+        if (!matchingSearchArticles.length) article.style.display = 'block';
+        else {
+          matchingSearchArticles.forEach((searchArticle) => {
+            if (article === searchArticle) {
+              searchArticle.style.display = 'block';
+            }
+          });
+        }
+      });
+    }
+
+    searchByAppNameInput.addEventListener('input', (e) => {
+      articles.forEach((article) => {
+        article.style.display = 'none';
+      });
+
+      const value = e.target.value.toLowerCase();
+      displayShowcase(value, filterValue)
+    });
+
+    searchByApiInput.addEventListener('click', (e) => {
+      const selections = document.querySelectorAll('li.button[selected]');
+      const filterPills = document.querySelectorAll('span.tag-pill');
+
+      const selectedOptions = []
+      selections.forEach((option) => {
+        return selectedOptions.push(option.textContent.trim())
+      })
+      filterPills.forEach((pill) => {
+        if (selectedOptions.includes(pill.textContent.trim())) {
+          const value = pill.textContent.trim().toLowerCase().split(' ').join('-');
+
+          if (!filterValue.includes(value)) {
+            filterValue.push(value);
+            const searchValue = searchByAppNameInput.value?.toLowerCase();
+            displayShowcase(searchValue, filterValue)
+          }
+
+          pill.removeAttribute('hidden');
+        }
+      })
+    });
+
+    const closeButtonPills = document.querySelectorAll('span.tag-pill button')
+    closeButtonPills.forEach(button => {
+      button.addEventListener('click', () => {
+        button.parentNode.setAttribute('hidden', true);
+
+        const enhancedSelect = document.querySelectorAll('.enhanced-select__options li');
+        enhancedSelect.forEach(option => {
+          if (option.textContent.trim() === button.parentNode.textContent.trim()) {
+            option.removeAttribute('selected');
+          }
+        });
+
+        const value = button.parentNode.textContent.trim().toLowerCase().split(' ').join('-');
+        filterValue = filterValue.filter(v => v !== value)
+        
+        const searchValue = searchByAppNameInput.value?.toLowerCase();
+        displayShowcase(searchValue, filterValue)
+      });
+    });
+
+    window.addEventListener('keydown', (e) => {
+      if (e.key === 'f' && e.metaKey) {
+        e.preventDefault();
+        searchByAppNameInput.focus();
+      }
+    });
+  </script>
```

Comment:
This should not be an inline script. Best sync with @johnrich85, who recently introduced new JS for the events page and can give you guidance.

Reply:
While doing so, also make sure to adhere to the style found elsewhere in the codebase: proper function naming, documented methods, ...